core/objlist.hpp

```diff
@@ -243,18 +249,34 @@ class ObjList : public ObjListBase {
         }
     }
 
-    // getter
     inline size_t get_sorted_size() const { return sorted_size; }
     inline size_t get_num_del() const { return objlist_data_.num_del_; }
     inline size_t get_hashed_size() const { return hashed_objs.size(); }
     inline size_t get_size() const override { return objlist_data_.get_size(); }
     inline size_t get_vector_size() const { return objlist_data_.get_vector_size(); }
     inline ObjT& get(size_t i) { return objlist_data_.data_[i]; }
 
+    bool write_to_disk() {
+        ASSERT_MSG(!list_name_.empty(), "Must have list name before writing to disk!");
+        DiskStore ds(list_name_);
+        BinStream bs;
+        bs << objlist_data_;
+        return ds.write(bs.get_buffer_vector());
+    }
+
+    void read_from_disk() {
+        ASSERT_MSG(!list_name_.empty(), "Must have list name before reading from disk!");
+        DiskStore ds(list_name_);
+        BinStream bs(std::move(ds.read()));
```

Comment:
`read` returns a const reference can not be moved, so actually I think the copy constructor will be invoked instead of the move constructor.